packages/components/organisms/f-checkout/src/services/tests/addressService.test.js

```diff
@@ -174,4 +174,124 @@ describe('addressService', () => {
+                    let store = {};
+                    return {
+                        getItem: function getItem (key) {
+                            return store[key];
+                        },
+                        setItem: function setItem (key, value) {
+                            store[key] = value.toString();
+                        },
+                        clear: function clear () {
+                            store = {};
+                        },
+                        removeItem: function removeItem (key) {
+                            delete store[key];
+                        }
+                    };
+                }());
+                Object.defineProperty(window, 'localStorage', { value: localStorageMock });
+            });
+
+            afterEach(() => {
+                window.localStorage.clear();
+                jest.resetAllMocks();
+            });
+
+            describe('when the address does NOT exist in local storage', () => {
+                it('should return false', () => {
+                    expect(addressService.isAddressInLocalStorage()).toBe(false);
+                });
+            });
+
+            describe('when the address does exist in local storage', () => {
+                it('should return true if there is a post code', async () => {
+                    window.localStorage.setItem('je-full-address-details', JSON.stringify(london3LinesDefault));
+                    expect(addressService.isAddressInLocalStorage()).toBe(true);
+                });
+
+                it('should return false if the customer only searched a postcode', async () => {
+                    window.localStorage.setItem('je-full-address-details', JSON.stringify({
+                        City: null,
+                        Field1: null,
+                        Field2: null,
+                        Line1: null,
+                        Line2: null,
+                        Line3: null,
+                        Line4: null,
+                        Line5: null,
+                        PostalCode: null,
+                        searchBoxAddress: 'BS3 4RL'
+                    }));
+                    expect(addressService.isAddressInLocalStorage()).toBe(false);
+                });
+            });
+        });
+
+        describe('if localStorage does NOT exist', () => {
+            beforeAll(() => {
+                Object.defineProperty(window, 'localStorage', { value: null });
+            });
+
+            afterAll(() => {
+                window.localStorage.clear();
+                jest.resetAllMocks();
+            });
+
+            it('should return false', () => {
+                expect(addressService.isAddressInLocalStorage()).toBe(false);
+            });
+        });
+    });
+
+    describe('getAddressFromLocalStorage ::', () => {
+        beforeEach(() => {
+            const localStorageMock = (function localStorageMock () {
```

Comment:
Probably worth pulling this bit out to a const at the top so it isn't duplicated

Reply:
Moved it into a test util file so it can be reused elsewhere. I know I have another task coming up which relies on local storage.